### PR TITLE
Converts pres_error bash script to ruby w/ tests

### LINKED ISSUE #77: (IR) need daily error report script, run from cron
@blalbrit  has spoken, and this cron, which used to be run on sul-sdr-services with old robot code, is desirable:

```
# Every day at 6am, noon and 6pm, email the ERRORS today in the SDR-PC logs
0 6,12,18 * * * $HOME/bin/sdr-preservation-core_errors.sh | /bin/mail -s '[sdr-pc] ingest errors today' sdr2service
```

- [ ] rewrite script in testable fashion (see weekly_status for a model)
- [ ] use whenever to schedule cron job ONCE a day (per @blalbrit) via code base
    - [ ] make sure the email gets to the right place (chase cron to ... to figure it out  and/or ask Ben/Tony)

### LINKED ISSUE #83: include whenever to add the cron job
add cron tasks in config/schedule.rb

### PR DESCRIPTION
**Pull request by SaravShah**
Should I add reasons for why I'm adding error_reporter to rubocop exceptions?

closes #77 
closes #83

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -64,9 +69,13 @@ Metrics/LineLength:
     - spec/preservation_ingest/verify_apo_spec.rb # one long line with fully namespaced classes
 Metrics/MethodLength:
   Exclude:
+    - lib/error_reporter.rb
     - robots/preservation_ingest/base.rb # execute_shell_command
     - robots/preservation_ingest/update_catalog.rb # one too many line [11/10]
 
+Metrics/PerceivedComplexity:
+  Exclude:
+    - lib/error_reporter.rb
```
Comment (ndushay) [inline .rubocop.yml]: blank line after this?

**Comment by SaravShah** [comment]
@ndushay ready for review again :)

File: lib/error_reporter.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    table_generator.each { |table| puts table.to_s unless table.rows.empty? }
+  end
+
+  def table_generator(log_files=default_log_files)
+    stack = []
+    today = Time.now.to_date.to_s
+
+    log_files.each do |file|
+      unless File.exist?(file) && File.size(file) > 0
+        puts "EMPTY or NON-EXISTANT: #{file}"
```
Comment (tallenaz) [inline lib/error_reporter.rb]: Typo here: `EXISTANT` should be `EXISTENT`

Comment (tallenaz) [reply lib/error_reporter.rb]: Otherwise LGTM

**Pull request merged**